mapzen-places-api/src/main/java/com/mapzen/places/api/internal/PeliasPlaceDetailFetcher.java

```diff
@@ -0,0 +1,81 @@
+package com.mapzen.places.api.internal;
+
+import com.mapzen.pelias.Pelias;
+import com.mapzen.pelias.gson.Feature;
+import com.mapzen.pelias.gson.Result;
+import com.mapzen.places.api.LatLng;
+import com.mapzen.places.api.LatLngBounds;
+import com.mapzen.places.api.Place;
+import com.mapzen.tangram.LngLat;
+
+import android.net.Uri;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static com.mapzen.places.api.internal.PlacePickerPresenterImpl.PROPERTY_NAME;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * Detail fetcher which uses pelias as the backing data source.
+ */
+public class PeliasPlaceDetailFetcher implements PlaceDetailFetcher {
+
+  Pelias pelias;
+  Feature feature;
+  LngLat coordinates;
+  Map<String, String> properties;
+
+  /**
+   * Constructs a new object.
+   */
+  public PeliasPlaceDetailFetcher() {
+    pelias = new Pelias();
+  }
+
+  @Override public void fetchDetails(LngLat coordinates, final Map<String, String> properties,
+      final OnPlaceDetailsFetchedListener listener) {
+    this.coordinates = coordinates;
+    this.properties = properties;
+    pelias.reverse(coordinates.latitude, coordinates.longitude, new Callback<Result>() {
+          @Override public void onResponse(Call<Result> call, Response<Result> response) {
+            String title = properties.get(PROPERTY_NAME);
+            for (Feature feature : response.body().getFeatures()) {
+              if (feature.properties.name.equals(title)) {
+                PeliasPlaceDetailFetcher.this.feature = feature;
+                String label = feature.properties.label;
+                label = label.replace(title + ",", "").trim();
+                listener.onPlaceDetailsFetched(title + "\n" + label);
```

Comment:
So essentially we are doing a reverse geocode using the coordinates of the POI embedded in the tile data then string matching the results by name to get place details from Pelias?

This seems less than ideal because it is
A. expensive - we are fetching more records than we need
B. brittle - we may not get an exact match between the tile `name` and Pelias `name`

An alternate approach would be to build the Pelias id using the tile data and make a request to the /place endpoint.

For example the tile https://tile.mapzen.com/mapzen/vector/v1/all/14/4823/6160.json contains the following POI data:

```
			"type": "Feature",
			"properties": {
				"kind": "building",
				"roof_material": "concrete",
				"name": "Five Nine John Lofts",
				"area": 1588,
				"sort_rank": 475,
				"kind_detail": "apartments",
				"addr_street": "John Street",
				"roof_shape": "flat",
				"height": 42.2,
				"volume": 67013,
				"source": "openstreetmap.org",
				"min_zoom": 14.0,
				"addr_housenumber": "59",
				"id": 278078952,
				"scale_rank": 4
			}
```

Using the `source` and `id` fields we can construct the following request to Pelias https://pelias.mapzen.com/v1/place?api_key=mapzen-ZTA7CTR&ids=openstreetmap:venue:way:278078952 which returns the correct venue data.

```
{
	"geocoding": {
		"version": "0.2",
		"attribution": "https://search.mapzen.com/v1/attribution",
		"query": {
			"ids": [{
				"source": "openstreetmap",
				"layer": "venue",
				"id": "way:278078952"
			}],
			"private": false
		},
		"engine": {
			"name": "Pelias",
			"author": "Mapzen",
			"version": "1.0"
		},
		"timestamp": 1483567579654
	},
	"type": "FeatureCollection",
	"features": [{
		"type": "Feature",
		"geometry": {
			"type": "Point",
			"coordinates": [-74.007235, 40.70904]
		},
		"properties": {
			"id": "way:278078952",
			"gid": "openstreetmap:venue:way:278078952",
			"layer": "venue",
			"source": "openstreetmap",
			"source_id": "way:278078952",
			"name": "Five Nine John Lofts",
			"housenumber": "59",
			"street": "John Street",
			"postalcode": "10038",
			"accuracy": "point",
			"country": "United States",
			"country_gid": "whosonfirst:country:85633793",
			"country_a": "USA",
			"region": "New York",
			"region_gid": "whosonfirst:region:85688543",
			"region_a": "NY",
			"county": "New York County",
			"county_gid": "whosonfirst:county:102081863",
			"locality": "New York",
			"locality_gid": "whosonfirst:locality:85977539",
			"borough": "Manhattan",
			"borough_gid": "whosonfirst:borough:421205771",
			"neighbourhood": "Financial District",
			"neighbourhood_gid": "whosonfirst:neighbourhood:85865711",
			"label": "Five Nine John Lofts, Manhattan, New York, NY, USA"
		},
		"bbox": [-74.007514, 40.70879, -74.007015, 40.709182]
	}],
	"bbox": [-74.007514, 40.70879, -74.007015, 40.709182]
}
```

However based on the [Pelias docs](https://github.com/pelias/pelias-doc/blob/master/place.md) ids should not be manually constructed.

> These gid strings should not be built manually, but rather used directly as-is to lookup additional details on the location that gid refers to.

We are using OSM data embedded in the tile to fetch addional OSM data from Pelias. There has got to be a better way to do this, right?